### DISCUSSION TITLE
im.relative_codelength_savings

### DISCUSSION THREAD

**Discussion by Brngm** [mapequation/infomap#340] [General]
Hello,
I have a graph G and a clustering C1 on the graph G

I calculated with Infomap the clustering C2 on the graph G
	with im.codelength I know the codelength of the graph G
	with im.relative_codelength_savings I know the relative-codelength-savings for C2

I want to know the relative-codelength-savings for C1

Does anyone know how to do it?

THANKS

**Comment by antoneri** [comment]
Hi!

Yes, you can do that using
```python
im = Infomap(silent=True, two_level=True)
# read graph somehow
# im.add_networkx_graph(G)
# or
# im.read_file("network.net")
# or add nodes and links manually
C1 = {
    # node: module
    1: 0,
    2: 0,
    3: 1,
    4: 1,
    # ...
}
im.initial_partition = C1
im.run(no_infomap=True)
print(im.relative_codelength_savings)
```

You can also pass the initial partition to `run`, see more here:
https://mapequation.github.io/infomap/python/infomap.html#infomap.Infomap.run

Hope this helps!

**Reply by Brngm** [reply]
Thanks a lot,

I just tested your code with my data
and it does exactly what I was looking for

have a good day

and well done! Infomap is one of the most elegant and efficient clustering method.

**Reply by antoneri** [reply]
Glad to be of help!
